### DISCUSSION TITLE
Add project statistics

### DISCUSSION THREAD

**Discussion by PavelTurk** [apache/netbeans#6777] [Ideas]
### Description

Currently NB doesn't provide any project statistics. At the same time IntelliJ has  [this one](https://plugins.jetbrains.com/plugin/4509-statistic) with 1,915,900 Downloads (I don't know if it's true). So, I suggest to add such feature to NB:

- must separate files by type (java, xml, css etc)
- must allow to exclude files/directories (node_modules, for example)
- must provide easy access to biggest/smallest files

### Are you willing to submit a pull request?

No

**Comment by mbien** [comment]
this is most likely covered by existing maven/gradle reporting plugins.

We also don't really track plugin requests on the NetBeans issue tracker -> ~closing~ edit converting to discussion.

**Comment by PavelTurk** [comment]
@mbien I find your position very strange.

**Comment by PavelTurk** [comment]
The reasons why I suggest this issue are the following:

1. Although there are different tools outside NB such as maven/gradle reporting plugins, [cloc](https://github.com/AlDanial/cloc) etc it is always better to have it inside IDE - it is both more convenient and gives more opportunities. For example - project -> right click -> statistics show biggest files, double click to open the file.
2. Such feature is very popular - cloc has 17k stars; IntelliJ statistics plugin 1,915,900 Downloads with v.1 release in 2009 and v.4.2.7 released in 2023; this [question about NB statistic support](https://stackoverflow.com/questions/10661636/how-to-count-lines-in-a-netbeans-project) on SO has 42k view and 35 upvotes.

**Comment by matthiasblaesing** [comment]
I also don't understand this swift close. We have several next-to useless reports in the tracker, for a feature request this looked fine. Whether this should be a plugin or not is a different question. We have serveral modules in the code base, that IMHO don't belong there (vendor specific integrations for example), this might or might not be implemented in the core.

**Comment by PavelTurk** [comment]
I would suggest to do this feature this way. Right click -> Project Statistics or main menu Source/Tools -> Project Statistics

After that there must be window with tabs. The first tab is general, all others are by file type - java, xml, etc.

General tab contains the following table, with all column header sorting:

```
                      file     file     total     blank     comment     code
Language              count    size     lines     lines     lines       lines
-------------------------------------------------------------------------------
Java                   100     200KB     1200      300       400        500   
XML                     20     100KB      600      100       200        300    
-------------------------------------------------------------------------------
Total:                 120     300KB     1800      400       600        800
```
Java tab contains the following table, with all column header sorting:

```
                       file     total     blank     comment     code
Language               size     lines     lines     lines       lines
-----------------------------------------------------------------------
ClassA                 10KB       100        20        30         50   
ClassB                 20KB       200        40        60        100
......   
-----------------------------------------------------------------------
Total:                200KB      1200       300       400        500
```

To make it more interesting it is possible to give percents, for example

```
ClassB                 20KB       200    40(20%)    60(30%)   100(50%)
```

**Reply by Chris2011** [reply]
Thx for your request and also for your idea. It is one implementation of my looooooong todo list. I would like to do it similiar to the IntelliJ plugin and I also don't see tje point of closing a feature request. I don't know the gradle/maven stuff. I also want this into NetBeans. Into the core or as a plugin I don't care.

**Comment by PavelTurk** [comment]
So, should I open a new issue?

**Reply by Chris2011** [reply]
We already have one. We should reopen it and retalk about it. IMHO.

**Reply by PavelTurk** [reply]
@Chris2011 I agree with you. @mbien Could you reopen the issue?

**Comment by PavelTurk** [comment]
Ok. I did everything I could. I am out. So, if anyone can reopen this issue let him do. Don't let @mbien kill this issue.

**Reply by Chris2011** [reply]
Please continue to be polite. As also we said, we can talk about creating a 3rd-party-plugin or direct into the core. What pros and cons we have of both, etc.

**Reply by PavelTurk** [reply]
@Chris2011 I am quite polite. But that's all. I am out.

**Comment by matthiasblaesing** [comment]
I reopened this as issue #6836 and tried to add the additional comments and improved description into the text.

@PavelTurk thank you for your idea. What might be important is, that reopening this as issue is no implication that it will be done. Chris already said that he eyed such a feature, but I assume, that for him the same is true for other contributors: There is only finite time and all contributors decide on their own where they put their work.
@Chris2011 you are a NetBeans PMC member. If you link your github account to your apache credentials you get super-foo powers in this repository too

**Comment by matthiasblaesing** [comment]
Ok - ignore my last comment. I closed the issue again. The lack of response from @mbien gave me the feeling he would not object.

**Reply by mbien** [reply]
https://github.com/apache/netbeans/issues/6836#issuecomment-1853310235
